bootstrap/kubeadm/internal/cloudinit/controlplane_init.go

```diff
@@ -31,9 +31,13 @@ const (
 {{.ClusterConfiguration | Indent 6}}
       ---
 {{.InitConfiguration | Indent 6}}
+-   path: /run/cluster-api/bootstrap
```

Comment:
this file is a placeholder to create the /run/cluster-api folder

Unfortunately `mkdir /run/cluster-api` doesn't work on Windows because it requires the filepath to use backward slashes. Instead of having to write different runcmd contents for Linux and Windows (which would also require detecting if the KubeadmConfig belongs to a linux or windows machine), this is a workaround to trigger the directory since cloudbase-init knows how to deal with `write_file`

Reply:
Can you elaborate on how/why this workaround is useful? I'm struggling to follow. At first, I thought that this was to contain the "success" file, but I see that this directory and the success file are siblings. And so, I'm failing to connect the dots. This directory is created regardless of success/failure of the bootstrap scripts...

Reply:
I think we should document this trick within a comment inline/before the template

Reply:
Maybe I should also rename the file to something more obvious / less confusing?

Reply:
What about `/run/.ClusterAPIBootstrapDone` ?

We shouldn't need to create a folder that way, and the file is hidden

Reply:
we'd still need to create /run, it's not present natively in Windows. Right now it happens to be created for /run/kubeadm, but I don't think we should rely on that assumption.